sklearn/utils/estimator_checks.py

```diff
@@ -384,16 +391,19 @@ def _construct_instance(Estimator):
             # For common test, we can enforce using `LinearRegression` that
             # is the default estimator in `RANSACRegressor` instead of `Ridge`.
             if issubclass(Estimator, RANSACRegressor):
-                estimator = Estimator(LinearRegression())
+                estimator = Estimator(_weighted(LinearRegression()))
```

Comment:
Will we eventually have versions here that raise if they don't receive the weights???

Reply:
we have decided that if metadata is requested but not provided, we don't raise or warn. We could have a setting to do this, but the discussion was that it's not where we want to bother the user.

Reply:
This could be a strict mode optionally set.

Reply:
The idea of allowing requested metadata to not be provided is to make them compatible with current kwargs